src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs

```diff
@@ -197,6 +182,75 @@ public async Task<bool> LaunchTestHostAsync(TestProcessStartInfo testHostStartIn
         };
     }
 
+    private string GetTestHostName(Architecture architecture, Framework targetFramework, PlatformArchitecture processArchitecture)
+    {
+        // We ship multiple executables for testhost that follow this naming schema:
+        // testhost<.tfm><.architecture>.exe
+        // e.g.: testhost.net472.x86.exe -> 32-bit testhost for .NET Framework 4.7.2
+        //
+        // The tfm is omitted for .NET Framework 4.5.1 testhost.
+        // testhost.x86.exe -> 64-bit testhost for .NET Framework 4.5.1
+        //
+        // The architecture is omitted for 64-bit (x64) testhost.
+        // testhost.net472.exe -> 64-bit testhost for .NET Framework 4.7.2
+        // testhost.exe -> 64-bit testhost for .NET Framework 4.5.1
+        //
+        // These omissions are done for backwards compatibility because orinally there were
+        // only testhost.exe and testhost.x86.exe, both built against .NET Framework 4.5.1.
+
+        StringBuilder testHostProcessName = new("testhost");
+
+        if (targetFramework.Name.StartsWith(".NETFramework,Version=v"))
+        {
+            // Transform target framework name into moniker.
+            // e.g. ".NETFramework,Version=v4.7.2" -> "net472".
+            var targetFrameworkMoniker = "net" + targetFramework.Name.Replace(".NETFramework,Version=v", string.Empty).Replace(".", string.Empty);
+
+            var isSupportedTargetFramework = SupportedTargetFrameworks.Contains(targetFrameworkMoniker);
+            if (isSupportedTargetFramework)
+            {
+                testHostProcessName.Append('.').Append(targetFrameworkMoniker);
+            }
+            else
+            {
+                // The .NET Framework 4.5.1 testhost that does not have moniker in the name is used as fallback.
+            }
+        }
+
+        var processArchitectureAsArchitecture = processArchitecture switch
+        {
+            PlatformArchitecture.X86 => Architecture.X86,
+            PlatformArchitecture.X64 => Architecture.X64,
+            PlatformArchitecture.ARM => Architecture.ARM,
+            PlatformArchitecture.ARM64 => Architecture.ARM64,
+            PlatformArchitecture.S390x => Architecture.S390x,
+            _ => throw new NotSupportedException(),
```

Comment:
Could it be worth adding a message or at least the `PlatformArchitecture` so we know what we don't support?

Reply:
This is probably not going to get reached if we run on new platform, I don't want to add a new message that needs to be translated. We should rather put all these translations between architecture objects into a common place and throw from there.